### PR TITLE
Add Netgear R7000 and R6400 Command Injection Auxiliary Module

### PR DESCRIPTION
**Pull request by cbrnrd**
@wvu-r7 Addresses #7698. This module executes a user-supplied shell command on Netgear R7000 and R6400 routers running firmware version V1.0.7.2_1.1.93 and possibly lower (to my knowledge and testing, anything > V1.0.4.30_1.1.67 but my environment is limited). The module could use some improvements (especially in `check()`) in terms of fingerprint accuracy. The method I used in the module is from [EDB-40889.](https://www.exploit-db.com/exploits/40889/)

## Verification

```
msf > use auxiliary/scanner/http/netgear_r7000_command_injection
msf auxiliary(netgear_r7000_command_injection) > options

Module options (auxiliary/scanner/http/netgear_r7000_command_injection):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   CMD                       yes       Command line to execute
   Proxies                   no        A proxy chain of format type:host:port[,type:host:port][...]
   RHOST                     yes       The remote target address
   RPORT    80               yes       The target port
   SSL      false            no        Negotiate SSL/TLS for outgoing connections
   VHOST                     no        HTTP server virtual host

msf auxiliary(netgear_r7000_command_injection) > set CMD echo 'cowsay'
CMD => echo cowsay
msf auxiliary(netgear_r7000_command_injection) > set RHOST 192.168.1.1
RHOST => 192.168.1.1
msf auxiliary(netgear_r7000_command_injection) > check
[*] 192.168.1.1:80 The target service is running, but could not be validated.  <--- This needs to be improved
msf auxiliary(netgear_r7000_command_injection) > run

[*] Sending request to 192.168.1.1
[*] Auxiliary module execution completed
msf auxiliary(netgear_r7000_command_injection) > 
```

### Other notes

I like the idea from @nixawk in #7698, but I simply don't have time (finals) to write a payload-server model like he suggested, but I'll try to make it in the time I have.

### CODE REVIEW

**Comment by wvu-r7** [comment]
@thecarterb: It this your first submitted module? Do you want me to review this normally or send you a PR with fixes?

**Comment by wvu-r7** [comment]
@thecarterb: Is this your first submitted module? Do you want me to review this normally or send you a PR with fixes?

**Comment by thecarterb** [comment]
@wvu-r7 Yeah it's my first one. A PR with fixes would be great, but if you want to review it normally that's fine too.

File: modules/auxiliary/scanner/http/netgear_r7000_command_injection.rb [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    register_options(
+      [
+        OptString.new('RHOST', [true, 'The remote target address', nil]),
+        OptString.new('CMD',   [true, 'Command line to execute', nil ]),
+      ], self.class)
+    end
+
+#taken from Dolibarr login utility, not really sure if it will work or not
+#Requests the login page which discloses the hardware, if it's an R7000 or R6400, return Detected
+def check
```
Comment (nixawk) [inline modules/auxiliary/scanner/http/netgear_r7000_command_injection.rb]: incorrect indentation

Comment (wvu-r7) [inline modules/auxiliary/scanner/http/netgear_r7000_command_injection.rb]: @nixawk: Are you continuing with review? I am planning to send a patch.

Comment (nixawk) [inline modules/auxiliary/scanner/http/netgear_r7000_command_injection.rb]: @wvu-r7 Please patch the pr.

**Comment by h00die** [comment]
@thecarterb Congrats on your first submission!  It can be a little daunting the first time with all the reviews and changes.  No worries!
Would it be possible to throw together a documentation on this?  I think its important since many dont have the hardware to test on.
A good reference would be here: https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/exploit/linux/http/ipfire_bashbug_exec.md and template here: https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/module_doc_template.md

**Comment by thecarterb** [comment]
@h00die Yeah sure no problem. I can add documentation and patch the pr with the suggested fixes my the end of the day today (probably earlier)

**Review by cbrnrd — COMMENTED** [review]
Fix indentation

**Comment by thecarterb** [comment]
@h00die Documentation should be good now

**Comment by h00die** [comment]
awesome, docs look good to me!

**Comment by wvu-r7** [comment]
I have some sweeping changes to make, so please hang tight. Thanks.

**Comment by thecarterb** [comment]
@wvu-r7 Any updates?

**Comment by wvu-r7** [comment]
@thecarterb: I will get to it soon. Your module is next on my priority list.

**Comment by thecarterb** [comment]
Great, thanks

**Comment by h00die** [comment]
@wvu-r7 your priority list is epic.  However, this has also been sitting for more than 4 weeks

**Comment by wvu-r7** [comment]
Yes.

**Comment by thecarterb** [comment]
@wvu-r7 @h00die Is it okay if i close this, make some improvements and then reopen? (I nuked my previous branch so I can't make edits any more)

**Comment by wvu-r7** [comment]
Sure, since I'm completely failing at making code changes, I'll do review on the new PR like we did for your last PR? :)

One quick note: this should really be an exploit, and if you have to, you can set the encoder to `cmd/ifs`. It's better to specify badchars, though.

Cheers!

**Pull request closed without merging**